gamestonk_terminal/stocks/discovery/disc_controller.py

```diff
@@ -150,57 +146,516 @@ def call_quit(self, _):
+            "-n",
+            "--num",
+            action="store",
+            dest="num",
+            type=check_non_negative,
+            default=5,
+            help="Number of past days to look for IPOs.",
+        )
+        parser.add_argument(
+            "--export",
+            choices=["csv", "json", "xlsx"],
+            default="",
+            type=str,
+            dest="export",
+            help="Export dataframe data to csv,json,xlsx file",
+        )
+        try:
+            if other_args:
+                if "-" not in other_args[0]:
+                    other_args.insert(0, "-n")
+
+            ns_parser = parse_known_args_and_warn(parser, other_args)
+            if not ns_parser:
+                return
+
+            finnhub_view.past_ipo(
+                num_days_behind=ns_parser.num,
+                export=ns_parser.export,
+            )
+
+        except Exception as e:
+            print(e, "\n")
+
+    def call_fipo(self, other_args: List[str]):
+        """Process fipo command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="fipo",
+            description="""
+                Future IPOs dates. [Source: https://finnhub.io]
+            """,
+        )
+        parser.add_argument(
+            "-n",
+            "--num",
+            action="store",
+            dest="num",
+            type=check_non_negative,
+            default=5,
+            help="Number of future days to look for IPOs.",
+        )
+        parser.add_argument(
+            "--export",
+            choices=["csv", "json", "xlsx"],
+            default="",
+            type=str,
+            dest="export",
+            help="Export dataframe data to csv,json,xlsx file",
+        )
+        try:
+            if other_args:
+                if "-" not in other_args[0]:
+                    other_args.insert(0, "-n")
+
+            ns_parser = parse_known_args_and_warn(parser, other_args)
+            if not ns_parser:
+                return
+
+            finnhub_view.future_ipo(
+                num_days_ahead=ns_parser.num,
+                export=ns_parser.export,
+            )
+
+        except Exception as e:
+            print(e, "\n")
 
     def call_gainers(self, other_args: List[str]):
         """Process gainers command"""
-        yahoo_finance_view.gainers_view(other_args)
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="gainers",
+            description="Print up to 25 top ticker gainers. [Source: Yahoo Finance]",
+        )
+        parser.add_argument(
+            "-n",
+            "--num",
+            action="store",
+            dest="num",
+            type=int,
+            default=5,
+            choices=range(1, 25),
+            help="Number of the top gainers stocks to retrieve.",
```

Comment:
Can we use that one type function I have that checks n is between two numbers?  Using the range as choices looks bad in the `-h` command?